src/components/Annotation.tsx

```diff
@@ -49,22 +61,40 @@ export default function Annotation({
     };
   }, [map, latitude, longitude]);
 
+  // Padding
+  useEffect(() => {
+    if (!annotation) return;
+    annotation.padding = new mapkit.Padding(paddingTop, paddingRight, paddingBottom, paddingLeft);
+  }, [annotation, paddingTop, paddingRight, paddingBottom, paddingLeft]);
+
+  // AnchorOffset
+  useEffect(() => {
+    if (!annotation) return;
+    annotation.anchorOffset = new DOMPoint(anchorOffsetX, anchorOffsetY);
+  }, [annotation, anchorOffsetX, anchorOffsetY]);
+
   // Simple values properties
   const properties = {
     title,
     subtitle,
     accessibilityLabel,
 
+    size,
+
     selected,
     animates,
     appearanceAnimation,
     draggable,
     enabled,
+    visible,
+    clusteringIdentifier,
   };
   Object.entries(properties).forEach(([propertyName, prop]) => {
     useEffect(() => {
       if (!annotation) return;
       // @ts-ignore
+      if (prop === undefined) { delete annotation[propertyName]; return; }
```

Comment:
Did you have a specific reason why you needed to use this instead of setting the property to `undefined`? It should be fine, but I’m worrying about this causing bugs with the other properties that are already supported by mapkit-react.

Reply:
Honestly, I really don't remember, since it is over a month ago already. I will investigate and let you know, thank you for your feedback and also the other valuable feedback you gave!

Reply:
So, I had a look into it. And the idea behind it was actually, that the provided property at some point might not be provided anymore. In this behavior I would expect the marker behave the same like if it never got that prop.

Setting the property to `undefined`, does set its value to `undefined`, meaning, that the marker does not fallback to a fallback value which it might have and has the behavior of an property with the value `undefined` instead which might be different.

No idea if this scenario though was a real aspect or just something I made up and thought would be clever. I also don't know if what I am telling really makes sense or there is a reason why it might be even stupid and produce bugs instead of preventing them. 

```
const marker = {};

// {}

marker['Property']='Some Value';

// {Property: "Some Value"}

marker['Property']=undefined;

// {Property: undefined}

delete(marker['Property']);

// {}
```

What do you think after this additional explanation about it?

Reply:
Thank you for your reply! I checked the details and it is probably fine to keep using `delete`. I was worried about MapKit JS preferring `{ property: undefined }` to `{}`, but it doesn’t seem to be the case.

For most of the properties, it won’t matter at all since we provide default values (hence it will never be `undefined`). The only times we set `undefined` as the default value are:
* `selectedGlyphImage`, but it should probably be `null` instead of `undefined` (it doesn’t seem to be documented, but this is the default value I get when trying it live)
* `selected`: the default value should be changed to `false` to match the behavior of MapKit JS (https://developer.apple.com/documentation/mapkitjs/annotationconstructoroptions/2991168-selected)
* `size`: this is the most complicated one. When setting `size` to null or undefined, MapKit JS throws an error. When using `delete`, it seems to silently ignore it. I’m not sure what we should do in this case, do you have any opinion on the subject?

Reply:
I feel like my solution is good, as long as you do not intentionally want a feature to have the `undefined` instead of the default value. Could you think of any case where you would want to set a feature `undefined`, while it's default is not `undefined`?

Also we could further investigate, what value to pass to the size in order to not throw this error. However, both an empty object `{}` or an object with `{width: undefined, height: undefined}` or `{width: null, height: null}` seem not to have the desired effect either and I wasn't able to find further information about the default value.

We could make this implementation conditional, however, if we cannot think about any implementation where we would want to explicitly set an prop to undefined instead of the default value I would most likely keep this implementation rather simple.

Reply:
I agree, we shouldn’t treat a property set to `undefined` differently than a missing property, and should use the default value in both cases.

The current behavior for `size` seems fine. We can merge it as it works currently. I’ll change the issues that I mentioned above with the default values of `selectedGlyphImage` and `selected` in a separate PR.

Reply:
Thank you so much! I updated my comment after I posted it, let's maybe discuss the last part deeper in the separate PR?